src/host/ut_host/ClipboardTests.cpp

```diff
@@ -67,90 +67,103 @@ class ClipboardTests
-
-        // since we're not in line selection, spaces should be trimmed from the end
-        tempPtr = text[0].data();
-        tempPtr += selection[0].right - selection[0].left - 2;
-        tempPtr++;
-        VERIFY_IS_NULL(wcsrchr(tempPtr, L' '));
-
-        // final line of selection should not contain CR/LF
-        tempPtr = text[3].data();
-        tempPtr += text[3].size();
-        tempPtr -= 2;
-        VERIFY_ARE_NOT_EQUAL(String(tempPtr), String(L"\r\n"));
+        const auto text = SetupRetrieveFromBuffer(false);
+
+        std::wstring expectedText;
+
+        // Block selection:
+        // - Add line breaks on wrapped and non-wrapped rows.
+        // - No trimming of trailing whitespace because trimming in Block
+        //   selection is disabled.
+
+        // All rows:
+        // First 15 columns selected -> 7 characters (9 columns) + 6 spaces
+        // Add CL/RF (except the last row)
+
+        // row 0
+        expectedText += L"AB\u304bC\u304dDE      ";
+        expectedText += L"\r\n";
+
+        // row 1
+        expectedText += L"AB\u304bC\u304dDE      ";
+        expectedText += L"\r\n";
+
+        // row 2
+        expectedText += L"AB\u304bC\u304dDE      ";
+        expectedText += L"\r\n";
+
+        // row 3
+        // last row -> no CL/RF
+        expectedText += L"AB\u304bC\u304dDE      ";
+
+        VERIFY_ARE_EQUAL(expectedText, text);
     }
-#pragma prefast(pop)
 
     TEST_METHOD(TestRetrieveLineSelectionFromBuffer)
     {
         // NOTE: This test requires innate knowledge of how the common buffer text is emitted in order to test all cases
         // Please see CommonState.hpp for information on the buffer state per row, the row contents, etc.
-
-        std::vector<til::inclusive_rect> selection;
-        const auto text = SetupRetrieveFromBuffers(true, selection);
-
-        // row 2, no wrap
-        // no wrap row before the end should have CR/LF
-        auto tempPtr = text[2].data();
-        tempPtr += text[2].size();
-        tempPtr -= 2;
-        VERIFY_ARE_EQUAL(String(tempPtr), String(L"\r\n"));
-
-        // no wrap row should trim spaces at the end
-        tempPtr = text[2].data();
-        VERIFY_IS_NULL(wcsrchr(tempPtr, L' '));
-
-        // row 1, wrap
-        // wrap row before the end should *not* have CR/LF
-        tempPtr = text[1].data();
-        tempPtr += text[1].size();
-        tempPtr -= 2;
-        VERIFY_ARE_NOT_EQUAL(String(tempPtr), String(L"\r\n"));
-
-        // wrap row should have spaces at the end
-        tempPtr = text[1].data();
-        auto ptr = wcsrchr(tempPtr, L' ');
-        VERIFY_IS_NOT_NULL(ptr);
+        const auto text = SetupRetrieveFromBuffer(true);
+
+        std::wstring expectedText;
+
+        // Line Selection:
+        // - Add line breaks on non-wrapped rows.
+        // - Trim trailing whitespace on non-wrapped rows.
+
+        // row 0
+        // no wrap -> trim trailing whitespace, add CL/RF

```

Comment:
nit all: replace `CL/RF` with `CR/LF` :smile:

Reply:
I must've gone through it a hundred times, but didn't notice this... shame on me 🤦‍♂️😂